source/Octopus.Cli/Util/FeatureDetectionExtensions.cs

```diff
@@ -7,22 +7,12 @@ public static class FeatureDetectionExtensions
     {
         public static bool SupportsChannels(this IOctopusAsyncRepository repository)
         {
-            return repository?.Client?.RootDocument.SupportsChannels() == true;
+            return repository?.Client?.HasLink("Channels") == true;
         }
-
-        public static bool SupportsChannels(this RootResource source)
-        {
-            return source?.HasLink("Channels") == true;
-        }
-
+        
         public static bool SupportsTenants(this IOctopusAsyncRepository repository)
         {
-            return repository?.Client?.RootDocument.SupportsTenants() == true;
-        }
-
-        public static bool SupportsTenants(this RootResource source)
-        {
-            return source?.HasLink("Tenants") == true;
+            return repository?.Client?.HasLink("Tenants") == true;
         }
 
         public static bool UsePostForChannelVersionRuleTest(this RootResource source)
```

Comment:
As we are going to remove `SupportsChannels(this RootResource source)`, should we remove this one or add `UsePostForChannelVersionRuleTest(this SpaceRootResource source)`?

Reply:
No this one still makes sense. We are using the root resource here because we need to know about the version, which only exists on the root resource.